### PR TITLE
[AAE-10779] User info component refactor

### PR DESCRIPTION
**Pull request by DiogoABastos**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
User info component handles content-services and process-services logic.
https://alfresco.atlassian.net/browse/AAE-10779

**What is the new behaviour?**
User info component has been refactored into 3 presentational components: ecm-user-info, bpm-user-info and identity-user-info. This was done to remove content-services and process-services logic from core

**Does this PR introduce a breaking change?** (check one with "x")

> - [x] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...
user-info component no longer exists in core and a new `userinfocomponent` needs to be created on alfresco-apps which aggregates the 3 presentational components in the adf repo.
**Other information**:

### CODE REVIEW

File: lib/process-services/src/lib/bpm-user-info/bpm-user-info.component.html [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                    </ng-template>
+                </div>
+            </ng-template>
+        </div>
+    </button>
+    <mat-menu #menu="matMenu" id="user-profile-lists" [xPosition]="menuPositionX" [yPosition]="menuPositionY"
+              [overlapTrigger]="false" class="adf-userinfo-menu">
+        <mat-tab-group id="tab-group-env" (click)="stopClosing($event)" selectedIndex="0" role="menuitem"
+                       class="adf-userinfo-tab" [class.adf-hide-tab]="!ecmUser">
+                       <mat-tab id="ecm-panel" label="{{ 'USER_PROFILE.TAB.CS' | translate }}" role="dialog"
```
Comment (DenysVuika) [inline lib/process-services/src/lib/bpm-user-info/bpm-user-info.component.html:40]: the id `ecm-panel` seems to be no longer used anywhere, could be removed?

File: lib/content-services/src/lib/ecm-user-info/ecm-user-info.component.html [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                </div>
+            </ng-template>
+        </div>
+    </button>
+
+    <mat-menu #menu="matMenu" id="user-profile-lists" [xPosition]="menuPositionX" [yPosition]="menuPositionY"
+              [overlapTrigger]="false" class="adf-userinfo-menu">
+        <mat-tab-group id="tab-group-env" (click)="stopClosing($event)" selectedIndex="0" role="menuitem"
+                       class="adf-userinfo-tab adf-hide-tab">
+            <mat-tab id="ecm-panel" label="{{ 'USER_PROFILE.TAB.CS' | translate }}" role="dialog"
```
Comment (DenysVuika) [inline lib/content-services/src/lib/ecm-user-info/ecm-user-info.component.html:43]: not used anymore in the unit/e2e tests, from what I see

Comment (DiogoABastos) [reply lib/content-services/src/lib/ecm-user-info/ecm-user-info.component.html:43]: Are you referring to the ecm-panel tab?

Comment (DenysVuika) [reply lib/content-services/src/lib/ecm-user-info/ecm-user-info.component.html:43]: yes, I checked the ADF, and no code references this id

Comment (DiogoABastos) [reply lib/content-services/src/lib/ecm-user-info/ecm-user-info.component.html:43]: I will remove both references to this id (ecm and bpm components)

File: docs/content-services/components/ecm-user-info.component.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+---
+Title: Ecm User Info component
+Added: v5.1.0
```
Comment (alep85) [inline docs/content-services/components/ecm-user-info.component.md:3]: Why `v5.1.0`? shouldn't be `v.6.0.0`?

File: docs/core/components/identity-user-info.component.md [hunk 4]
```diff
@@ -1,18 +1,18 @@
 ---
-Title: User Info component
-Added: v2.0.0
+Title: Identity User Info component
+Added: v5.1.0
```
Comment (alep85) [inline docs/core/components/identity-user-info.component.md:3]: The same as above

File: docs/process-services/components/bpm-user-info.component.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+---
+Title: Bpm User Info component
+Added: v5.1.0
```
Comment (alep85) [inline docs/process-services/components/bpm-user-info.component.md:3]: The same as above

File: lib/content-services/src/lib/ecm-user-info/ecm-user-info.component.html [hunk 6]
```diff
@@ -94,43 +77,11 @@
...
-                </mat-card>
-            </mat-tab>
-            <mat-tab id="identity-panel" role="dialog" *ngIf="mode==='SSO'">
-                <mat-card class="adf-userinfo-card" *ngIf="identityUser$ | async as identityUser">
+            <mat-tab id="identity-panel" role="dialog" *ngIf="mode === userInfoMode.CONTENT_SSO">
+                <mat-card class="adf-userinfo-card">
                     <mat-card-header class="adf-userinfo-card-header"
                                      [style.background-image]="'url(' + bpmBackgroundImage + ')'">
-                        <div *ngIf="ecmUser$ | async as ecmUser">
+                        <div>
```
Comment (alep85) [inline lib/content-services/src/lib/ecm-user-info/ecm-user-info.component.html:84]: Do we still need that `div`?

File: docs/content-services/components/ecm-user-info.component.md (hunk 3 above)
Comment (DiogoABastos) [reply docs/content-services/components/ecm-user-info.component.md:3]: I will update that

**Review by alep85 — APPROVED** [review]

**Review by eromano — APPROVED** [review]

File: demo-shell/src/app/components/app-layout/user-info/user-info.component.ts [hunk 7]
```diff
@@ -52,23 +34,12 @@ export class UserInfoComponent implements OnInit, OnDestroy {
...
-
+    mode: UserInfoMode;
     ecmUser$: Observable<EcmUserModel>;
     bpmUser$: Observable<BpmUserModel>;
     identityUser$: Observable<IdentityUserModel>;
     selectedIndex: number;
-    private destroy$ = new Subject();
+    userInfoMode = UserInfoMode;
 
     constructor(private peopleContentService: PeopleContentService,
```
Comment (eromano) [inline demo-shell/src/app/components/app-layout/user-info/user-info.component.ts:44]: user-info.component.ts should not depend from  peopleContentService and bpmUserService to be generic

**Review by eromano — CHANGES_REQUESTED** [review]

**Review by eromano — APPROVED** [review]

**Pull request merged**